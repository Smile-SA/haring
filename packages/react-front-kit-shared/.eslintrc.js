module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
