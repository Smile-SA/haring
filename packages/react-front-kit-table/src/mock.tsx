/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
