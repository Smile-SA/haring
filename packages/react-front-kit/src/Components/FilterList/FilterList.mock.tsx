import type { IFilter } from '@smile/react-front-kit-shared';

import { Select, TextInput } from '@mantine/core';

export const filtersMock: IFilter[] = [
  {
    active: false,
    component: <TextInput placeholder="some value" />,
    id: 1,
    label: 'Filter A',
  },
  {
    active: false,
    component: (
      <Select
        data={[
          { label: 'Some value', value: 'some value' },
          {
            label: 'Another value',
            value: 'another value',
          },
        ]}
        value="some value"
      />
    ),
    id: 2,
    label: 'Filter B',
  },
  {
    active: true,
    component: <TextInput placeholder="some value" />,
    id: 3,
    label: 'Filter C',
  },
  {
    active: true,
    component: (
      <Select
        data={[
          { label: 'Another value', value: 'another value' },
          { label: 'Some value', value: 'some value' },
        ]}
        value="another value"
      />
    ),
    id: 4,
    label: 'Filter D',
  },
];
