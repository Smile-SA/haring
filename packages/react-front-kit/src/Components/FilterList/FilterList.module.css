.filterComponent {
  min-width: 184px;
}

.manageFilterModalBody {
  padding: 24px;
}

.manageFilterModalHeader {
  margin-bottom: 16px;
  padding: 24px 24px 0 24px;
}

.manageFilterModalTitle {
  font-size: 18px;
  font-weight: 700;
}

.manageFiltersButton {
  align-items: start;
  background-color: transparent;
  border: none;
}

.manageFiltersLabel {
  gap: 4px;
}

.submitButton {
  min-width: 184px;
}
