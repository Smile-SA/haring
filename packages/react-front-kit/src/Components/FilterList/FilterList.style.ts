import { createStyles } from '@mantine/core';

export const useStyles = createStyles(() => ({
  filterComponent: {
    minWidth: 184,
  },
  manageFilterModalBody: {
    padding: '24px',
  },
  manageFilterModalHeader: {
    marginBottom: '16px',
    padding: '24px 24px 0 24px',
  },
  manageFilterModalTitle: {
    fontSize: '18px',
    fontWeight: 700,
  },
  manageFiltersButton: {
    alignItems: 'start',
    backgroundColor: 'transparent',
    border: 'none',
  },
  manageFiltersLabel: {
    gap: 4,
  },
  submitButton: {
    minWidth: 184,
  },
}));
