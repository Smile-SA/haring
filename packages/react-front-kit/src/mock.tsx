/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
