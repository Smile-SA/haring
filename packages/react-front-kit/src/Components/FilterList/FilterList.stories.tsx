import type { Meta, StoryObj } from '@storybook/react';

import { action } from '@storybook/addon-actions';

import { FilterList as Cmp } from './FilterList';
import { filtersMock } from './FilterList.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/FilterList',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const FilterList: IStory = {
  args: {
    filters: filtersMock,
    onActiveFiltersChange: action('Active filters changed'),
    onSubmit: action('Filters submitted'),
  },
};

export const ColumnDirection: IStory = {
  args: {
    direction: 'column',
    filters: filtersMock,
    onActiveFiltersChange: action('Active filters changed'),
    onSubmit: action('Filters submitted'),
    spacing: 12,
  },
};
