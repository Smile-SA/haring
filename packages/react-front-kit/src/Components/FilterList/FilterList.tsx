'use client';

import type { ModalProps, StackProps } from '@mantine/core';
import type { IFilter } from '@smile/react-front-kit-shared';
import type { ElementType, ReactElement } from 'react';

import { Button, Group, Modal, Stack, useMantineTheme } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import { Plus } from '@phosphor-icons/react';
import { useState } from 'react';

import { SearchableCheckboxList } from '../SearchableCheckboxList/SearchableCheckboxList';

import { useStyles } from './FilterList.style';

export interface IFilterListProps
  extends Omit<StackProps, 'justify' | 'onSubmit'> {
  direction?: 'column' | 'row';
  filters: IFilter[];
  filtersManageLabel?: string;
  manageFilterModalSearchPlaceholder?: string;
  manageFilterModalSearchSubmit?: string;
  manageFilterModalTitle?: string;
  modalProps?: ModalProps;
  onActiveFiltersChange?: (filters: IFilter[]) => void;
  onSubmit?: (activeFilters: IFilter[]) => void;
  submitLabel?: string;
}

export function FilterList(props: IFilterListProps): ReactElement {
  const {
    direction = 'row',
    filters,
    filtersManageLabel = 'Manage filters',
    manageFilterModalSearchPlaceholder = 'Search in filters',
    manageFilterModalSearchSubmit = 'Confirm changes',
    manageFilterModalTitle = 'Manage filters',
    modalProps,
    onActiveFiltersChange,
    onSubmit,
    submitLabel = 'Filter',
    ...containerProps
  } = props;
  const [localFilters, setLocalFilters] = useState<IFilter[]>(filters);
  const [manageFiltersModal, handleManageFiltersModal] = useDisclosure(false);
  const Container: ElementType = direction === 'row' ? Group : Stack;

  const theme = useMantineTheme();
  const { classes } = useStyles();

  function handleSubmit(): void {
    onSubmit?.(localFilters.filter((filter) => filter.active));
  }

  function handleManageFiltersSubmit(activeFilters: IFilter[]): void {
    handleManageFiltersModal.close();

    const newFilters = localFilters.map((filter, i) =>
      activeFilters[i].id === filter.id ? activeFilters[i] : filter,
    );
    onActiveFiltersChange?.(activeFilters);
    setLocalFilters(newFilters);
  }

  return (
    <Container spacing={10} {...containerProps}>
      {localFilters
        .filter((filter) => filter.active)
        .map((filter) => (
          <span key={filter.id} className={classes.filterComponent}>
            {filter.component}
          </span>
        ))}
      {onSubmit ? (
        <Button className={classes.submitButton} onClick={handleSubmit}>
          {submitLabel}
        </Button>
      ) : null}
      <Button
        className={classes.manageFiltersButton}
        classNames={{ label: classes.manageFiltersLabel }}
        onClick={handleManageFiltersModal.open}
        variant="default"
      >
        <Plus color={theme.colors.dark[6]} size={12} />
        {filtersManageLabel}
      </Button>
      <Modal
        centered
        classNames={{
          body: classes.manageFilterModalBody,
          header: classes.manageFilterModalHeader,
          title: classes.manageFilterModalTitle,
        }}
        onClose={handleManageFiltersModal.close}
        opened={manageFiltersModal}
        size="md"
        title={manageFilterModalTitle}
        {...modalProps}
      >
        <SearchableCheckboxList<IFilter>
          buttonLabel={manageFilterModalSearchSubmit}
          checkboxes={filters}
          onClickButton={handleManageFiltersSubmit}
          placeholder={manageFilterModalSearchPlaceholder}
        />
      </Modal>
    </Container>
  );
}
