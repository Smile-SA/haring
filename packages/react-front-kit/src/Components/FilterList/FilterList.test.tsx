import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { FilterList } from './FilterList';
import { filtersMock } from './FilterList.mock';

describe('FilterList', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <FilterList filters={filtersMock} />,
    );
    expect(container).toMatchSnapshot();
  });
});
