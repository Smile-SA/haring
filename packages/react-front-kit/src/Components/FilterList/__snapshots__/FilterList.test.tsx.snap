// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList matches snapshot 1`] = `
<div>
  <div
    class="mantine-Group-root mantine-8k3bl3"
  >
    <span
      class="mantine-1eieff6"
    >
      <div
        class="mantine-InputWrapper-root mantine-TextInput-root mantine-q6oiko"
      >
        <div
          class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-1v7s5f8"
        >
          <input
            aria-invalid="false"
            class="mantine-Input-input mantine-TextInput-input mantine-1qswl1p"
            id="mantine-f4bipx4bi"
            placeholder="some value"
            type="text"
          />
        </div>
      </div>
    </span>
    <span
      class="mantine-1eieff6"
    >
      <div
        class="mantine-InputWrapper-root mantine-Select-root mantine-q6oiko"
      >
        <div
          aria-controls="mantine-f4bipx4bi"
          aria-expanded="false"
          aria-haspopup="listbox"
          class=""
          role="combobox"
          tabindex="-1"
        >
          <input
            type="hidden"
            value="another value"
          />
          <div
            class="mantine-Input-wrapper mantine-Select-wrapper mantine-1v7s5f8"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocomplete="off"
              class="mantine-Input-input mantine-Select-input mantine-kkodi1"
              data-mantine-stop-propagation="false"
              id="mantine-f4bipx4bi"
              readonly=""
              type="search"
              value="Another value"
            />
            <div
              class="mantine-1qj7q0z mantine-Input-rightSection mantine-Select-rightSection"
            >
              <svg
                data-chevron="true"
                fill="none"
                style="color: rgb(134, 142, 150); width: 1.125rem; height: 1.125rem;"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </span>
    <button
      class="mantine-UnstyledButton-root mantine-Button-root mantine-13kxxue"
      data-button="true"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-Button-label mantine-1bj6mez"
        >
          <svg
            fill="#25262b"
            height="12"
            viewBox="0 0 256 256"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224,128a8,8,0,0,1-8,8H136v80a8,8,0,0,1-16,0V136H40a8,8,0,0,1,0-16h80V40a8,8,0,0,1,16,0v80h80A8,8,0,0,1,224,128Z"
            />
          </svg>
          Manage filters
        </span>
      </div>
    </button>
  </div>
</div>
`;
